presto-main/src/main/java/com/facebook/presto/operator/NativeExecutionOperator.java

```diff
@@ -0,0 +1,219 @@
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * TODO: The lifecycle of NativeExecutionOperator is:
+ *  1. Launch the native engine external process when initialing the operator.
+ *  2. Serialize and pass the planFragment, tableWriteInfo and taskSource to the external process through NativeExecutionTask APIs.
+ *  3. Block on NativeExecutionTask's pollResult call to retrieve the data in {@link SerializedPage} page format back external process.
+ *  4. Deserialize the serialized page to presto Page and return it back to driver from the getOutput method.
+ *  5. The close() will be called by the driver when the pollResult returns empty and the Future of NativeExecutionTask has completed.
+ *  6. Shut down the external process upon calling of close() method
+ */
+public class NativeExecutionOperator
+        implements SourceOperator
+{
+    private final PlanNodeId sourceId;
+    private final OperatorContext operatorContext;
+    private final LocalMemoryContext systemMemoryContext;
+    // TODO: will be used to create the {@link NativeExecutionTask}
+    private final PlanFragment planFragment;
+    // TODO: will be used to create the {@link NativeExecutionTask}
+    private final TableWriteInfo tableWriteInfo;
+    // TODO: will be used to deserialize the  {@link SerializedPage}
+    private final PagesSerde serde;
+    private TaskSource taskSource;
+    private boolean finished;
+
+    public NativeExecutionOperator(
+            PlanNodeId sourceId,
+            OperatorContext operatorContext,
+            PlanFragment planFragment,
+            TableWriteInfo tableWriteInfo,
+            PagesSerde serde)
+    {
+        this.sourceId = requireNonNull(sourceId, "sourceId is null");
+        this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
+        this.systemMemoryContext = operatorContext.localSystemMemoryContext();
+        this.planFragment = requireNonNull(planFragment, "planFragment is null");
+        this.tableWriteInfo = requireNonNull(tableWriteInfo, "tableWriteInfo is null");
+        this.serde = requireNonNull(serde, "serde is null");
+    }
+
+    @Override
+    public OperatorContext getOperatorContext()
+    {
+        return operatorContext;
+    }
+
+    @Override
+    public boolean needsInput()
+    {
+        return false;
+    }
+
+    @Override
+    public void addInput(Page page)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * TODO: The getOutput method will call NativeExecutionTask APIs to:
+     *  1. Submit the plan to the external process
+     *  2. Call pollResult method to get latest buffered result.
+     *  3. Call getTaskInfo method to get the TaskInfo and propagate it
+     *  4. Deserialize the {@link SerializedPage} page to presto Page and return it back
```

Comment:
Deserialize the polled {@link SerializedPage} to {@link Page} and return it back